### PR TITLE
Harden mule signing task

### PR DESCRIPTION
**Pull request by btoll**
## Summary

Improved the signing shell script in the following ways:
1. Fixed gpg warning by setting correct permissions in the docker container.
1. Instead of assuming the build status file is present, check for it first.

## Test Plan

`mule -f package-signing package-signing`

### CODE REVIEW

File: scripts/release/mule/sign/sign.sh [hunk 1]
```diff
@@ -56,8 +64,12 @@ shasum -a 512 *.tar.gz *.deb *.rpm >> "$HASHFILE"
...
 gpg -u "$SIGNING_KEY_ADDR" --clearsign "$HASHFILE"
 
-gpg -u "$SIGNING_KEY_ADDR" --clearsign "$STATUSFILE"
-gzip -c "$STATUSFILE.asc" > "$STATUSFILE.asc.gz"
+# The status file *should* be present, but let's check to be safe.
+if [ -f "$STATUSFILE" ]
+then
+    gpg -u "$SIGNING_KEY_ADDR" --clearsign "$STATUSFILE"
+    gzip -c "$STATUSFILE.asc" > "$STATUSFILE.asc.gz"
+fi
```
Comment (egieseke) [inline scripts/release/mule/sign/sign.sh:72]: Is it considered an error if the status file does not exist?  Should this print an error message and return an error code if it is missing?

Comment (onetechnical) [reply scripts/release/mule/sign/sign.sh:72]: I think there's a separate bug where the status file is getting put in the wrong place. The file is not required and not published, so I think this is a workaround; alternatively we could fix the placement of the status file and nip this in the bud.

File: scripts/release/mule/sign/sign.sh [hunk 2]
```diff
@@ -47,7 +55,7 @@ do
     gpg -u rpm@algorand.com --detach-sign "$file"
 done
 
-HASHFILE=hashes_${CHANNEL}_${OS_TYPE}_${ARCH_TYPE}_${VERSION}
+HASHFILE="hashes_${CHANNEL}_${OS_TYPE}_${ARCH_TYPE}_${VERSION}"
```
Comment (bricerisingalgorand) [inline scripts/release/mule/sign/sign.sh:58]: Will this sign all of the packages for different architectures in tmp/node_dir, or just the ones matching the host computers cpu architecture?

File: scripts/release/mule/sign/sign.sh (hunk 1 above)
Comment (btoll) [reply scripts/release/mule/sign/sign.sh:72]: Yes, optimally the build status file is put back in its original location.

File: scripts/release/mule/sign/sign.sh (hunk 2 above)
Comment (btoll) [reply scripts/release/mule/sign/sign.sh:58]: The latter.

File: scripts/release/mule/sign/sign.sh (hunk 1 above)
Comment (btoll) [reply scripts/release/mule/sign/sign.sh:72]: I'm going to remove that conditional.

**Review by onetechnical — APPROVED** [review]
This looks good to me, but will require fixing build status location. #1668

**Review by egieseke — APPROVED** [review]
Looks good.

File: scripts/release/mule/sign/sign.sh (hunk 2 above)
Comment (bricerisingalgorand) [reply scripts/release/mule/sign/sign.sh:58]: Cool, for this one we can run this for each os/arch we'll need. Later on maybe we can update this like the package scripts to iterate on all available files

**Review by bricerisingalgorand — APPROVED** [review]

**Pull request merged**